src/app/code/community/Afterpay/Afterpay/Helper/Data.php

```diff
@@ -411,8 +411,11 @@ public function giftCardsPlaceOrder()
      */
     public function getChunkedBody($response){
         try{
-            return $response->getBody();
+            $body = $response->getBody();
         }catch(\Exception $e){
+            // We only want to ignore the exception in this specific case.
+            if($e->getMessage() !== "Error parsing body - doesn't seem to be a chunked message") throw $e;
```

Comment:
What type of error does this throw? if it is specific enough you are always better of using type comparison then the string, as Magento has string localisation which will not be caught by this string comparison.

Reply:
It throws a `Zend_Http_Exception`, which is used for quite a few different things.  The code throwing the error ( `throw new Zend_Http_Exception("Error parsing body - doesn't seem to be a chunked message");` doesn't seem to use localisation?

Reply:
From other PR: should we rethink whether or not we want catch Exception, if we know for sure Afterpay consistently return response non chunked we can just go ahead and decode it

Reply:
:+1: as long as it was considered

Reply:
and if the response's header doesn't match the response body it's a bug on Afterpay side that need to be brought up with them.

Reply:
Agree @willtran that it should be brought up with them.  I think I want to leave it there so that in the event they fix their server, the code continues to work.

@michaelwylde  - if you're ok with this, can you please approve and merge?

Reply:
logged as issue afterpay#27